source/common/grpc/grpc_web_filter.cc

```diff
@@ -0,0 +1,108 @@
+#include "common/grpc/grpc_web_filter.h"
+
+#include <arpa/inet.h>
+
+#include "common/common/base64.h"
+#include "common/http/headers.h"
+
+namespace Envoy {
+namespace Grpc {
+
+const uint8_t GrpcWebFilter::GRPC_WEB_TRAILER = 0b10000000;
+
+GrpcWebFilter::GrpcWebFilter() : is_text_request_(false), is_text_response_(false) {}
+
+GrpcWebFilter::~GrpcWebFilter() {}
+
+// Implements StreamDecoderFilter.
+Http::FilterHeadersStatus GrpcWebFilter::decodeHeaders(Http::HeaderMap& headers, bool) {
```

Comment:
IMO it would be great to get some more comments on the transforms that are going on, since I think most people are not familiar with grpc-web. It might be sufficient to have a top level doc comment in the .h file per my other comment.